### PR TITLE
ESLint JSDoc

### PR DESCRIPTION
**Pull request by 0xCLARITY**
## High Level Overview of Change

Fixes a bunch of JSDoc things now that `eslint-plugin-jsdoc` is actually enabled.

### Context of Change

This upgrades us to the latest version of `@xpring-eng/eslint-config-base` and its associated peer dependencies, including a newly enabled `elint-plugin-jsdoc`.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Everything is now correctly JSDoc'ed.

## Test Plan

`npm run test` is still green.

### CODE REVIEW

File: src/Common/utils.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import * as addressCodec from 'ripple-address-codec'
 /**
  * A prefix applied when hashing a signed transaction blob.
  *
- * @see https://xrpl.org/basic-data-types.html#hashes
+ * @see {@link https://xrpl.org/basic-data-types.html#hashes|XRPL Data Types - Hashes}.
```
Comment (hbergren) [inline src/Common/utils.ts:8]: Again, TSDoc likes to pair `@see` tags with `@link` tags to clarify what is going on.

File: src/Common/utils.ts [hunk 2]
```diff
@@ -96,10 +97,10 @@ const utils = {
   /**
    * Decode a `ClassicAddress` from a given x-address.
    *
-   * @see https://xrpaddress.info/
+   * @see {@link https://xrpaddress.info/|XRPL X-Address Format}.
    *
    * @param xAddress - The xAddress to decode.
-   * @returns A `ClassicAddress`
+   * @returns A `ClassicAddress`.
```
Comment (hbergren) [inline src/Common/utils.ts:103]: We require full sentences (Start with a capital letter, end with a period, more or less).

File: src/XRP/wallet-factory.ts [hunk 3]
```diff
@@ -1,22 +1,30 @@
-import Wallet from './wallet'
 import Utils from '../Common/utils'
 
+import Wallet from './wallet'
+
 /**
  * Encapsulates various methods for generating Wallets.
  */
 export default class WalletFactory {
+  public readonly isTest: boolean
```
Comment (hbergren) [inline src/XRP/wallet-factory.ts:9]: We disallow parameter properties on classes, because being more explicit is easier for people new to TypeScript (and ES6 classes don't have parameter properties).

File: src/XRP/wallet.ts [hunk 4]
```diff
@@ -27,8 +22,13 @@ export interface WalletGenerationResult {
  * A wallet object that has an address and keypair.
  */
 class Wallet {
-  private readonly publicKey: string
-  private readonly privateKey: string
+  /**
+   * The default derivation path to use with BIP44.
+   */
+  public static defaultDerivationPath = "m/44'/144'/0'/0/0"
```
Comment (hbergren) [inline src/XRP/wallet.ts:28]: Make this a static property on the class so I don't have to document `getDefaultDerivationPath()`.

File: src/XRP/wallet.ts [hunk 5]
```diff
@@ -27,8 +22,13 @@ export interface WalletGenerationResult {
...
+
+  public readonly publicKey: string
+  public readonly privateKey: string
```
Comment (hbergren) [inline src/XRP/wallet.ts:31]: Make these two properties public so I can get rid of their `getter()` functions.

They are marked `readonly`, so they are effectively `private` with `getter()` functions.

File: test/XRP/wallet.test.ts [hunk 6]
```diff
@@ -19,7 +19,7 @@ const derivationPathTestCases = {
       '0090802A50AA84EFB6CDB225F17C27616EA94048C179142FECF03F4712A07EA7A4',
     expectedMainNetAddress: 'XVMFQQBMhdouRqhPMuawgBMN1AVFTofPAdRsXG5RkPtUPNQ',
     expectedTestNetAddress: 'TVHLFWLKvbMv1LFzd6FA2Bf9MPpcy4mRto4VFAAxLuNpvdW',
-    messageHex: Buffer.from('test message', 'utf-8').toString('hex'),
+    messageHex: Buffer.from('test message', 'utf8').toString('hex'),
```
Comment (hbergren) [inline test/XRP/wallet.test.ts:22]: Totally unrelated, I just like `utf8` better (and when I look at the `Buffer` documentation, it uses `utf8` as the example string, not `utf-8`.

**Comment by hbergren** [comment]
The PR is a little bigger than I would have liked, so sorry in advance if it's hard to parse!

**Review by keefertaylor — APPROVED** [review]

**Review by amiecorso — APPROVED** [review]
